samples/graph-activity-feed-broadcast/nodejs/server.js

```diff
@@ -0,0 +1,242 @@
+const polly = require('polly-js');
+
+var delegatedToken = "";
+var applicationToken = "";
+var localdata = [];
+app.use(express.static(path.join(__dirname, 'static')));
+app.engine('html', require('ejs').renderFile);
+app.set('view engine', 'ejs');
+app.set('views', __dirname);
+const ENV_FILE = path.join(__dirname, '.env');
+require('dotenv').config({ path: ENV_FILE });
+
+// parse application/json
+app.use(express.json());
+
+app.get('/broadcast', function (req, res) {
+  var tenantId = process.env.TenantId;
+  auth.getAccessToken(tenantId).then(async function (token) {
+    applicationToken = token;
+    var requestData = localdata;
+    res.render('./views/BroadcastNotification', { data: JSON.stringify(requestData) });
+  });
+});
+
+app.get('/BroadcastDetails', function (req, res) {
+  var requestId = req.url.split('=')[1];
+  let requestData = {};
+  if (requestId != null) {
+    localdata.map(item => {
+      if (item.id == requestId) {
+        requestData = item;
+      }
+    })
+  }
+  res.render('./views/BroadcastDetails', { data: JSON.stringify(requestData) });
+});
+
+// Pop-up dialog to ask for additional permissions, redirects to AAD page
+app.get('/auth-start', function (req, res) {
+  res.render('./views/auth-start', { clientId: JSON.stringify(process.env.ClientId) });
+});
+
+// End of the pop-up dialog auth flow, returns the results back to parent window
+app.get('/auth-end', function (req, res) {
+  var clientId = process.env.ClientId;
+  res.render('./views/auth-end', { clientId: clientId });
+});
+
+// On-behalf-of token exchange
+app.post('/auth/token', function (req, res) {
+  var tid = req.body.tid;
+  var token = req.body.token;
+  var scopes = ["https://graph.microsoft.com/User.Read"];
+
+  // Creating MSAL client
+  const msalClient = new msal.ConfidentialClientApplication({
+    auth: {
+      clientId: process.env.ClientId,
+      clientSecret: process.env.ClientSecret
+    }
+  });
+
+  var oboPromise = new Promise((resolve, reject) => {
+    msalClient.acquireTokenOnBehalfOf({
+      authority: `https://login.microsoftonline.com/${tid}`,
+      oboAssertion: token,
+      scopes: scopes,
+      skipCache: true
+    }).then(result => {
+      delegatedToken = result.accessToken
+      resolve();
+    }).catch(error => {
+      reject({ "error": error.errorCode });
+    });
+  });
+
+  oboPromise.then(function (result) {
+    res.json(result);
+  }, function (err) {
+    console.log(err); // Error: "It broke"
+    res.json(err);
+  });
+});
+
+// Send notification to group chat for task creation.
+app.post('/SendNotificationToOrganisation', (req, res) => {
+  var taskDetails = {
+    id: req.body.id,
+    title: req.body.title,
+    description: req.body.description,
+    createdBy: req.body.userName,
+    userId: req.body.userId
+  };
+
+  localdata.push(taskDetails);
+  var appId;
+
+  axios.get("https://graph.microsoft.com/v1.0/users/" + req.body.userId + "/teamwork/installedApps/?$expand=teamsAppDefinition", {
```

Comment:
Could you please use the Graph SDK wherever possible?

Reply:
updated code with graph sdk method